operator/Dockerfile.cross

```
@@ -0,0 +1,33 @@
+ARG BUILD_IMAGE=grafana/loki-build-image:0.18.0
+
+FROM golang:1.17.9-alpine as goenv
+RUN go env GOARCH > /goarch && \
+  go env GOARM > /goarm
+
+FROM --platform=linux/amd64 $BUILD_IMAGE as builder
+COPY --from=goenv /goarch /goarm /
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
+COPY internal/ internal/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -o manager main.go
```

I don't know too much about cross platform builds, but as I understood, `GOARCH` and `GOARM` environment variables must be provided.
At least the values for them are copied into the Dockerfile in line 8.

```suggestion
RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on GOARCH=$(cat /goarch) GOARM=$(cat /goarm) go build -a -o manager main.go
```